### DISCUSSION TITLE
scheduling

### DISCUSSION THREAD

**Discussion by unicomp21** [esfx/esfx#55] [Q&A] [answered]
Is there some way to do typescript deferred scheduling based on time slicing via performance.now() in the browser?  the target use case is video games, and runtime generation of assets using idle cpu.  Perhaps a package already exists which I'm not aware of?

@rbuckton

**Comment by rbuckton** [comment, marked as answer]
I imagine there are some existing APIs that could potentially help, such as [`window.requestIdleTimeout`](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestIdleCallback), but this isn't an area I'm actively investigating.
